packages/notifier-seeder/src/models/catalogItemEventNotification.ts

```diff
@@ -120,3 +79,65 @@ export type CatalogItemV1Notification = Omit<
   mode: string;
   attributes?: CatalogAttributesV1Notification;
 };
+
+// CatalogItemV1AddedV1 | ClonedCatalogItemV1AddedV1 | CatalogItemV1UpdatedV1 | MovedAttributesFromEserviceToDescriptorsV1
+export type CatalogItemNotification = {
+  catalogItem?: CatalogItemV1Notification;
+};
+
+// CatalogItemWithDescriptorsDeletedV1
+export type CatalogItemDescriptorDeletedNotification = {
+  catalogItem?: CatalogItemV1Notification;
+  descriptorId: string;
+};
+
+export type CatalogItemDocumentUpdateNotification = {
+  eServiceId: string;
+  descriptorId: string;
+  documentId: string;
+  updatedDocument?: CatalogDocumentV1Notification;
+  serverUrls: string[];
+};
+
+// CatalogItemDocumentAddedV1
+export type CatalogItemDocumentAddedNotification = {
+  eServiceId: string;
+  descriptorId: string;
+  document?: CatalogDocumentV1Notification;
+  isInterface: boolean;
+  serverUrls: string[];
+};
+
+// CatalogItemDescriptorAddedV1 | CatalogItemDescriptorUpdatedV1
+export type CatalogDescriptorNotification = {
+  eServiceId: string;
+  catalogDescriptor?: CatalogDescriptorV1Notification;
+};
+
+// CatalogItemRiskAnalysisAddedV1 | CatalogItemRiskAnalysisUpdatedV1 | CatalogItemRiskAnalysisDeletedV1;
+export type CatalogItemRiskAnalysisNotification = {
+  catalogItem?: CatalogItemV1Notification;
+  catalogRiskAnalysisId: string;
+};
+
+// CatalogItemDeletedV1 | CatalogItemDocumentUpdatedV1
+export type CatalogItemIdNotification = {
+  catalogItemId: string;
+};
+
+// CatalogItemDocumentDeletedV1
+export type CatalogItemDocumentDeletedNotification = {
+  eServiceId: string;
+  descriptorId: string;
+  documentId: string;
+};
+
+export type CatalogItemEventNotification =
+  | CatalogItemNotification
+  | CatalogItemIdNotification
+  | CatalogItemDocumentDeletedNotification
+  | CatalogItemDocumentUpdateNotification
+  | CatalogItemDocumentAddedNotification
+  | CatalogDescriptorNotification
+  | CatalogItemDescriptorDeletedNotification
+  | CatalogItemRiskAnalysisNotification;
```

Comment:
This type looks the same as before but with the different shapes of the union defined as standalone types and not inline. What is the advantage we gain in terms of type-checking? 

Is there a way to have these models not defined manually here but generated from the previous Scala codebase/protobuf events? It could help us avoid typos/errors here and when we have to redefine similar models for other services e.g., agreement.

Reply:
As discussed offline, these types are necessary because you cannot use the type generated by the protobuf definition as-is.
The legacy scala code generates notification messages with a JSON schema different from the protobuf definition, in fact during serialization the fields with type `Date` and `Enum` are generated with a different type from the generated protobuf file, in detail scala library "JSON Spray" handle those in different manner.,